### PR TITLE
Create a setup method for config and logger to reduce boilerplate

### PR DESCRIPTION
**Pull request by triarius**
This should be used at the start of all cli commands to parse their config and set up the logger (which depends on the config)

The function `setupLoggerAndConfig` does this. It has the signature
```go
// setupLoggerAndConfig populates the given config struct with values from the
// CLI flags and environment variables. It returns the config struct, a logger
// based on the resulting config, a reference to the config file (if any), and
// a function, which must be deferred.
//
// Presently, the returned function will wind down the profiler, which is only
// optionally started, based on the config. However, it may be extended in the
// future to clean up other resources. Importantly, the calling code does not
// need to know or care about what the returned function does, only that it
// must defer it.
func setupLoggerAndConfig[T any](c *cli.Context, opts ...configOpts) (
	cfg T,
	l logger.Logger,
	f *cliconfig.File,
	done func(),
) {
```
The upshot is that we get to replace the following boilderplate with 1 line:
```diff
-		// The configuration will be loaded into this struct
-		cfg := ArtifactDownloadConfig{}
-
-		loader := cliconfig.Loader{CLI: c, Config: &cfg}
-		warnings, err := loader.Load()
-		if err != nil {
-			fmt.Printf("%s", err)
-			os.Exit(1)
-		}
-
-		l := CreateLogger(&cfg)
-
-		// Now that we have a logger, log out the warnings that loading config generated
-		for _, warning := range warnings {
-			l.Warn("%s", warning)
-		}
-
-		// Setup any global configuration options
-		done := HandleGlobalFlags(l, cfg)
+		cfg, l, _, done := setupLoggerAndConfig[ArtifactDownloadConfig](c)
		defer done()
```

I've used generics to avoid having to type assert the returned config. At the moment, the type constraint is `any`, but I think it makes sense to restrict this to an interface with methods returning the global config fields. Then we won't need to use reflection to determine if the `--debug` flag is set, for example. However, I have refrained from doing that here.

### CODE REVIEW

**Review by moskyb — APPROVED** [review]
looking awesome - one nit, but i'm very happy to merge this as is

File: clicommand/env_get.go [hunk 1]
```diff
@@ -62,7 +62,16 @@ Examples:
...
-type EnvGetConfig struct{}
+type EnvGetConfig struct {
+	Format string `cli:"format"`
+
+	// Global flags
+	Debug       bool     `cli:"debug"`
+	LogLevel    string   `cli:"log-level"`
+	NoColor     bool     `cli:"no-color"`
+	Experiments []string `cli:"experiment" normalize:"list"`
+	Profile     string   `cli:"profile"`
```
[the remark this answers is not in the document]
Comment (triarius) [reply clicommand/env_get.go:73]: Yeah, totally agree. My plan is to do that as part of this:

> At the moment, the type constraint is any, but I think it makes sense to restrict this to an interface with methods returning the global config fields. Then we won't need to use reflection to determine if the --debug flag is set, for example. However, I have refrained from doing that here.

[the remark this answers is not in the document]
Comment (moskyb) [reply clicommand/env_get.go:73]: ah! turns out i can't read :P

[the remark this answers is not in the document]
Comment (triarius) [reply clicommand/env_get.go:73]: Haha, or, I wasn't explicit enough.

**Pull request merged**